cli/src/mctl.rs

```diff
@@ -172,12 +177,13 @@ async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
         Sub::Publish {
             uuid,
             key,
+            protocol,
         } => serde_json::to_string_pretty(
             &call::<_, PublishNexusReply>(
                 &opt.socket,
                 "publish_nexus",
                 Some(json!({ "uuid": uuid,
-                    "share" : ShareProtocol::Nbd as i32,
+                    "share" : ShareProtocol::from_str(&protocol),
```

Comment:
Using ShareProtocol::None will result in an error mayastor/src/bdev/nexus/nexus_share.rs
So should the mctl even support specifying that value?

Reply:
For now, let this be. There is probably a case to remove ShareProtocol::None

Reply:
If ShareProtocol::None exists as a value, I think that mctl should support setting it. If nothing else, it lets you test the error  handling in nexus_share.rs.